### PR TITLE
feat: add spotless for Java

### PR DESCRIPTION
**Pull request by hsiliev**
Adds spotless for Java files with:
* eclpse java formatter
* remove unused imports
* import order java, javax, jakarta, com, org, net.atos
* type annotations formatting
* license header for **Lifely only**

Solves: PZ-1162, PZ-1163

### CODE REVIEW

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -40,6 +40,9 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Fetch origin main
+        run: git fetch origin main
```
Comment (edgarvonk) [inline .github/workflows/build-test-deploy.yml:44]: Why is this needed? What problem does this solve?

File: build.gradle.kts [hunk 2]
```diff
@@ -225,6 +224,35 @@ swaggerSources {
...
+
+    format("misc") {
+        target("*.gradle", ".gitattributes", ".gitignore", ".containerignore", ".dockerignore")
+
+        trimTrailingWhitespace()
+        indentWithSpaces()
+        endWithNewline()
+    }
+    java {
+        targetExclude("**/src/generated/**", "**/build/generated/**")
```
Comment (edgarvonk) [inline build.gradle.kts:239]: nice! that's still not all our generated files because they don't all end up in 'generated' packages right now but we can fix that. :-)

File: build.gradle.kts [hunk 3]
```diff
@@ -225,6 +224,35 @@ swaggerSources {
...
+
+        googleJavaFormat()
+
+        removeUnusedImports()
+        importOrder("java", "javax", "jakarta", "com", "org", "net.atos")
+
+        formatAnnotations()
+        licenseHeader("""
+/*
```
Comment (edgarvonk) [inline build.gradle.kts:248]: Does this mean all our Java license headers need to be like this?

**Review by edgarvonk — COMMENTED** [review]
So how does this work right now? For now it just generated warnings or something? During build time?

File: .github/workflows/build-test-deploy.yml (hunk 1 above)
Comment (hsiliev) [reply .github/workflows/build-test-deploy.yml:44]: I'm trying to avoid several commits with changes to all java files. Current settings seems ok, but I'm sure we'll be amending them one or two times until Google format fits what we want.

Therefore I added `ratchetFrom("origin/main")` below, so spotless will fail only for the newly changed files.
However as we do shallow clone without origin/main this fails.

File: build.gradle.kts (hunk 2 above)
Comment (hsiliev) [reply build.gradle.kts:239]: Yep - that's one of the reasons I don't want to have this applied to all files. I don't have enough knowledge yet to add the rest of the clients :(

File: build.gradle.kts (hunk 3 above)
Comment (hsiliev) [reply build.gradle.kts:248]: Yes. I believe we have the option to specify different targets and license headers for them, but that might make the config pretty bloated.

File: .github/workflows/build-test-deploy.yml (hunk 1 above)
Comment (edgarvonk) [reply .github/workflows/build-test-deploy.yml:44]: >I'm trying to avoid several commits with changes to all java files

To be honest I would not try to avoid that at all. Just make a huge PR with changes to all Java files. Why not?

Comment (hsiliev) [reply .github/workflows/build-test-deploy.yml:44]: Ok. I can do that

File: build.gradle.kts (hunk 2 above)
Comment (edgarvonk) [reply build.gradle.kts:236]: ah clear. I can fix those clients so that they are all in 'generated' packages I think.

File: build.gradle.kts (hunk 3 above)
Comment (edgarvonk) [reply build.gradle.kts:245]: Ok. I think it's probably a good idea anyway to have them all the same. :-)

File: build.gradle.kts [hunk 4]
```diff
@@ -225,6 +224,32 @@ swaggerSources {
...
+
+        googleJavaFormat().reflowLongStrings().skipJavadocFormatting()
+
+        removeUnusedImports()
+        importOrder("java", "javax", "jakarta", "com", "org", "net.atos")
+
+        formatAnnotations()
+        licenseHeader("""
+            /*
+             * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
```
Comment (bas-info-nl) [inline build.gradle.kts:246]: New files should not contain copyrights by "Atos". Existing files should keep header as is.

Comment (bas-info-nl) [reply build.gradle.kts:246]: Unless touched by us ofc.

File: src/main/java/net/atos/client/bag/BAGClientService.java [hunk 5]
```diff
@@ -37,55 +36,49 @@ public class BAGClientService {
 
     public static final String DEFAULT_CRS = "epsg:28992";
 
-    private static final String ADRES_EXPAND = "panden, adresseerbaarObject, nummeraanduiding, openbareRuimte, woonplaats";
+    private static final String ADRES_EXPAND =
```
Comment (edgarvonk) [inline src/main/java/net/atos/client/bag/BAGClientService.java:39]: I guess the max line length is now the default 120 right? For Kotlin we usually set it to 140. Is that maybe possible here too? I think 120 is a little conservative. ;-)

**Review by edgarvonk — APPROVED** [review]
Nice one!

File: src/main/java/net/atos/client/zgw/ztc/ZTCClientService.java [hunk 6]
```diff
@@ -383,16 +407,17 @@
...
+                            "URI '%s' does not start with value for environment variable "
+                                    + "'%s': '%s'",
+                            uri,
+                            ENV_VAR_ZGW_API_CLIENT_MP_REST_URL,
+                            configuratieService.readZgwApiClientMpRestUrl()));
         }
 
-        return JAXRSClientFactory.getOrCreateClient().target(uri)
+        return JAXRSClientFactory.getOrCreateClient()
+                .target(uri)
```
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/java/net/atos/client/zgw/ztc/ZTCClientService.java:420]: We should create a JIRA ticket for this (if it really is an issue) and ignore it here I think.

File: src/main/java/net/atos/client/bag/BAGClientService.java (hunk 5 above)
Comment (hsiliev) [reply src/main/java/net/atos/client/bag/BAGClientService.java:39]: I think it's even 80 chars by default and not possible to change.

A lot of things are not possible with google formatter:
* having parameters offset based on where the method parenthesis starts, so they are aligned pretty
* line length
* no configuration for lots of things (by design)

We might want to switch to Eclipse formatter but that means configuring it via Eclipse (not IntelliJ).

Comment (hsiliev) [reply src/main/java/net/atos/client/bag/BAGClientService.java:39]: https://github.com/diffplug/spotless/issues/134

**Review by bas-info-nl — APPROVED** [review]

File: src/main/java/net/atos/client/bag/BAGClientService.java (hunk 5 above)
Comment (edgarvonk) [reply src/main/java/net/atos/client/bag/BAGClientService.java:39]: Hmm, maybe you are right yes..

File: build.gradle.kts (hunk 4 above)
Comment (edgarvonk) [reply build.gradle.kts:246]: True.. That makes this quite tricky I think..

File: src/main/java/net/atos/client/bag/BAGClientService.java (hunk 5 above)
Comment (hsiliev) [reply src/main/java/net/atos/client/bag/BAGClientService.java]: We're now using 140 chars

File: build.gradle.kts [hunk 7]
```diff
@@ -225,6 +224,32 @@ swaggerSources {
...
+        indentWithSpaces()
+        endWithNewline()
+    }
+    java {
+        targetExclude("**/src/generated/**", "**/build/generated/**")
+
+        googleJavaFormat().reflowLongStrings().skipJavadocFormatting()
+
+        removeUnusedImports()
+        importOrder("java", "javax", "jakarta", "com", "org", "net.atos")
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply build.gradle.kts]: Import settings are now in `config/importOrder.txt` and pretty close to Eclipse/IntelliJ

File: build.gradle.kts (hunk 4 above)
Comment (hsiliev) [reply build.gradle.kts]: Lifely-only license files are updated. The rest are left alone and needs our attention.

File: src/main/java/net/atos/client/bag/BAGClientService.java [hunk 8]
```diff
@@ -99,7 +99,7 @@ public AdresIOHal readAdres(final URI adresURI) {
 
     private Invocation.Builder createInvocationBuilder(final URI uri) {
         return JAXRSClientFactory.getOrCreateClient().target(uri)
-                .request("application/hal+json", "application/problem+json")
-                .header(X_API_KEY, API_KEY);
+                                 .request("application/hal+json", "application/problem+json")
```
Comment (edgarvonk) [inline src/main/java/net/atos/client/bag/BAGClientService.java:102]: Wow, quite some indent. Oh well, I guess this is the standard then right?

Comment (edgarvonk) [reply src/main/java/net/atos/client/bag/BAGClientService.java:102]: Not sure I like these huge indents. Actually I don't. ;-)

**Review by edgarvonk — APPROVED** [review]
I guess it's ok but those big indents I don't fancy..

File: src/main/java/net/atos/client/bag/BAGClientService.java (hunk 8 above)
Comment (hsiliev) [reply src/main/java/net/atos/client/bag/BAGClientService.java:102]: That can be changed to an ident of 8 spaces if we agree on that.

File: src/main/java/net/atos/client/bag/api/AdresUitgebreidApi.java [hunk 9]
```diff
@@ -55,34 +81,50 @@
...
+     * een exacte match met één adres, bv. in geval van meerdere objecten met huisnummertoevoegingen en/of huisletters. Met de exacteMatch
+     * parameter kan worden aangegeven dat alleen object(en) die exact overeenkomen met de opgegeven parameters, geretourneerd moeten
+     * worden. 4. Zoek uitgebreide adres informatie van huidige adressen met een zoekterm. Voor paginering, zie: [functionele specificatie
+     * paginering](https://github.com/lvbag/BAG-API/blob/master/Features/paginering.feature). Als inclusiefEindStatus&#x3D;true, dan worden
+     * ook actuele adressen met een eind status geleverd, zie [functionele specificatie
+     * inclusiefEindstatus](https://github.com/lvbag/BAG-API/blob/master/Features/inclusief-eind-status.feature).
      */
     @GET
     @Produces({"application/hal+json", "application/problem+json"})
     public AdresUitgebreidHalCollection zoekAdresUitgebreid(@QueryParam("postcode") String postcode,
```
Comment (edgarvonk) [inline src/main/java/net/atos/client/bag/api/AdresUitgebreidApi.java:117]: I see it doesn't automatically move this `@QueryParam("postcode") String postcode,` onto the next line. A shame. That is what e.g. Detekt does for Kotlin and something I like much better. Guess we still have some manual work left then to improve the formatting.

File: src/main/java/net/atos/client/bag/BAGClientService.java (hunk 8 above)
Comment (hsiliev) [reply src/main/java/net/atos/client/bag/BAGClientService.java]: Done. @edgarvonk have a look at this variant

At least it's easily configurable via IntelliJ :)

File: src/main/java/net/atos/client/bag/api/AdresUitgebreidApi.java (hunk 9 above)
Comment (hsiliev) [reply src/main/java/net/atos/client/bag/api/AdresUitgebreidApi.java]: Changed
